src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -44,6 +45,8 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </remarks>
     internal abstract class CSharpSemanticModel : SemanticModel
     {
+        private static readonly ObjectPool<SegmentedList<ISymbol>> s_symbolListPool = new ObjectPool<SegmentedList<ISymbol>>(() => new SegmentedList<ISymbol>());
```

Comment:
For me, the gains claimed here aren't worth the complexity and risk brought in by another custom object pool. I believe this change can be reasonably reduced to:

1. Make `FilterNotReferencable` operate on `ArrayBuilder<T>` instead of `ImmutableArray<T>`
2. Make `FilterOverriddenOrHiddenIndexers` operate on `ArrayBuilder<T>` instead of `ImmutableArray<T>`

Reply:
Yeah, I'll switch this to not have the custom pool as it does complicate this change, and really only benefitted the cases where there were a large number of symbols.